### PR TITLE
More accurate parsing of the Mesos task info in SI tests MARATHON-7958

### PR DESCRIPTION
**Pull request by alenkacz**
Summary:
Some parts of mesos task info were queries out of first task status but that is not possible anymore because Mesos started sending TASK_STARTING status. We need to explicitly ask for STATUS_RUNNING for this to work.

JIRA issues: MARATHON-7958, MARATHON-7959

### CODE REVIEW

File: tests/system/marathon_pods_tests.py [hunk 1]
```diff
@@ -425,7 +425,7 @@ def test_pod_with_container_network():
 
     tasks = common.get_pod_tasks(pod_id)
 
-    network_info = tasks[0]['statuses'][0]['container_status']['network_infos'][0]
+    network_info = common.running_status_network_info(tasks[0]['statuses'])
```
Comment (zen-dog) [inline tests/system/marathon_pods_tests.py]: Same problem in line 457?

Comment (alenkacz) [reply tests/system/marathon_pods_tests.py]: Good point, thanks. Fixed

File: tests/system/common.py [hunk 2]
```diff
@@ -737,3 +737,13 @@ def marathon_leadership_changed(original_leader):
...
+
+
+def running_status_network_info(task_statuses):
+    """ From a given list of statuses retrieved from mesos API it returns network info of running task.
+    """
+    for task_status in task_statuses:
+        if task_status['state'] == "TASK_RUNNING":
+            return task_status['container_status']['network_infos'][0]
+
+    pytest.fail("did not found a TASK_RUNNING status in task statuses: %s" %(task_statuses,))
```
Comment (zen-dog) [inline tests/system/common.py]: I would just `assert False, "Haven't found..."` here

Comment (alenkacz) [reply tests/system/common.py]: I don't see much difference but changed...

File: tests/system/common.py [hunk 3]
```diff
@@ -737,3 +737,13 @@ def marathon_leadership_changed(original_leader):
...
+
+
+def running_status_network_info(task_statuses):
+    """ From a given list of statuses retrieved from mesos API it returns network info of running task.
+    """
+    for task_status in task_statuses:
+        if task_status['state'] == "TASK_RUNNING":
+            return task_status['container_status']['network_infos'][0]
+
+    assert False, "Did not found a TASK_RUNNING status in task statuses: %s" % (task_statuses,)
```
Comment (jeschkies) [inline tests/system/common.py]: Nit: "Did not find a TASK_RUNNING"

File: tests/system/common.py [hunk 4]
```diff
@@ -737,3 +737,13 @@ def marathon_leadership_changed(original_leader):
...
     __marathon_leadership_changed_in_marathon_api(original_leader)
     __marathon_leadership_changed_in_mesosDNS(original_leader)
+
+
+def running_status_network_info(task_statuses):
+    """ From a given list of statuses retrieved from mesos API it returns network info of running task.
+    """
+    for task_status in task_statuses:
+        if task_status['state'] == "TASK_RUNNING":
+            return task_status['container_status']['network_infos'][0]
```
Comment (jeschkies) [inline tests/system/common.py]: Somehow it seems to me that we make a lot of assumptions here. E.g. we retrieve just the first network info of the first task that is task_running. Shouldn't identify the task by name?

Comment (alenkacz) [reply tests/system/common.py]: I am querying tasks by name now, good point

File: tests/system/common.py [hunk 5]
```diff
@@ -737,3 +737,31 @@ def marathon_leadership_changed(original_leader):
...
+
+    assert False, "Did not find a TASK_RUNNING status in task statuses: %s" % (task_statuses,)
+
+
+def task_by_name(tasks, name):
+    """ Find mesos task by its name
+    """
+    for task in tasks:
+        print(task['name'])
+        print(name)
```
Comment (zen-dog) [inline tests/system/common.py]: Forgot to remove `print`s?

**Pull request merged**